scipy/special/_softmax.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import division
+import numpy as np
+from scipy.special import logsumexp
+
+
+def softmax(X, theta=1.0, axis=None):
+    r"""
+    Softmax function
+
+    Compute the softmax of each element along an axis of X.
+
+    .. math:: \sigma(X)_j = \frac{e^{\theta x_j}}{\sum_{k=1}^{K}e^{\theta x_k}} \textrm{ for } j = 1, \ldots, K
+
+    Parameters
+    ----------
+    X : array_like
+        Input array.
+    theta : float, optional
+            parameter, used as a multiplier prior to exponentiation. Default = 1.0
+    axis : int, optional
+           axis to compute values along. Default is the first non-singleton axis.
+
+    Returns
+    -------
+    s : ndarray
+        an array the same size as X. The result will sum to 1 along the specified axis.
+    """
+    X = np.asarray(X)
```

Comment:
This will prevent the function from working with duck arrays. Perhaps an `if not hasattr(X, '__array_ufunc__')`?